frontend/src/components/dropdowns/BookDropdown.tsx

```diff
@@ -13,7 +13,6 @@ export interface BookDropdownProps {
   bookTitlesList: string[]; // List of book titles
   selectedBook: string; // The selected book
   placeholder?: string; // Placeholder for the dropdown
-  refreshKey?: number; // Used for refreshing the dropdown, necessary for a workaround
```

Comment:
I am not sure why this was removed? Does the dropdown work without this?

Reply:
Yup, I figured out a better way to do it. I was actually not following good practice before, because I was directly updating the state variable instead of doing something like set___(...), which was causing the UI not to refresh. But the new library I used allows us to more easily do something like set__() for only one element in an array